### ISSUE TITLE
Mobile - Long blog posts titles overflow on mobile devices

### ISSUE THREAD

**Issue by AlinChindea** [neontribe/www#301]
Split from #247 

**Plan**
- [ ] check current `font-size` of a blog post title
- [ ] try out different `font-size` values 
- [ ] try `word-break` property
- [ ] try hyphenation

**Options**

1.  Keep current size - xlarge or 3.5rem or 63px and `word-break`
![blog_title_word-break_xlarge_font_size](https://user-images.githubusercontent.com/29477363/65321740-e7b09000-db9c-11e9-8c06-98cb4a45e292.png)

2. Smaller but still large font size (36px) and `word-break`
![36px](https://user-images.githubusercontent.com/29477363/65418011-55ee9000-ddf3-11e9-8f44-97f580d2532a.png)

or on a smaller device
![36px-smaller-device](https://user-images.githubusercontent.com/29477363/65418049-71f23180-ddf3-11e9-88a0-10bc9c50dcaa.png)

3. Medium font size - `word-break` would be useless here as `font-size` value is small enough to fit the title on the screen
a) example with medium font-size (22.5px or 1.25rem) and `word-break` property on
![blog_post_title_medium_font_size_word-break](https://user-images.githubusercontent.com/29477363/65321953-76251180-db9d-11e9-868d-8fe3d9434eb4.png)

b)example with medium font-size (22.5px or 1.25rem) and `word-break` property _**off**_
![blog_post_title_medium_font_size_no_word-break](https://user-images.githubusercontent.com/29477363/65322106-d025d700-db9d-11e9-8091-67d928a5b031.png)

**Comment by HedgehogHolly** [comment]
In terms of design, I definitely prefer option 2, I think it's more in proportion than Option 1. With option 3, it doesn't have enough difference in size for the header to be instantly recognisable as a title.

**Comment by AlinChindea** [comment]
Thanks, @HedgehogHolly, for your feedback! Just to add that the sizes are pre-defined and I didn't just make those up :-) However, I might have given you a larger size for that option 2 than it's the case. I wrote 45px but's in fact 36px (see below size scale and image with the right size for option2).  clearly, the header should be instantly recognisable

```
fontSizes = {
  small: '0.85rem', // 15.3px
  normal: '1rem', // 18px
  medium: '1.25rem', // 22.5px
  large: '2rem', // 36px
  xlarge: '3.5rem', //63 px
}

**Comment by falconx** [comment]
@AlinChindea nice comprehensive list of options :wink: I agree we want to persist the hierarchical relationship by keeping the title text large. Which doesn't leave us with too many options if we want to avoid overflow. I would go with the `word-break` approach and perhaps try to progressively enhanance with `hyphens` https://caniuse.com/#feat=mdn-css_properties_hyphens

**Comment by AlinChindea** [comment]
thanks @falconx! I was testing the `hyphens` property but it really doesn't work - at least not in Chrome and Firefox for Fedora 30 (latest version of the browsers). In Chrome, I get a warning that it's not supported and in Firefox it simply doesn't do what we expect it to do (even after I disabled the `word-break` property as the two don't get along). On MDN, it says:

> Hyphenation rules are language-specific. In HTML, the language is determined by the lang attribute, and browsers will hyphenate only if this attribute is present and if an appropriate hyphenation dictionary is available.

Further, the title of the blog post is actually included in the markdown file like so: 
```
---
id: 97
uuid: 'e0df3890-28e9-43de-9646-fee831f8d9cb'
title: 'Crunchbangplusplus 9 tweaks'
slug: 'crunchbangplusplus-9-tweaks'
...
---
```
Sure, we can add HTML tags in there but that won't be the most elegant option. There is a small JS library that can allow us to do this but it might be an overkill for our purposes. 

I'll go with the `word-break` for now but any advice is more than welcome.

**Comment by falconx** [comment]
Thanks for looking into it further!

I agree we don't want to be touching the content in the markdown files and that adding a JavaScript library would be overkill.

Here's a really good, recent, article about hyphen support https://justmarkup.com/articles/2019-01-28-a-look-at-css-hyphenation-in-2019

We could progressively enhance so that hyphens work where they're supported but using `word-break` alone is sufficient enough :)

**Comment by AlinChindea** [comment]
Thank you for sharing the article! I've read his 2015 post - that he references in the one you've shared - and was wondering why there's no newer posts about it. Guess I missed it in the results. I found [this one useful, too](http://clagnut.com/blog/2395/ ) (pays off sometimes to click on the second page of Google search results :))). 

> You should keep in mind that while it works in Chromium-based browsers on Mac & Android platforms it doesn’t work at the moment (January 2019) on Windows and Linux.

This explains why it wasn't working on my Chrome: I'm on a Linux distro (Fedora 30)... Meanwhile, I've tested on my Android phone and the `hyphens` work fine. However, we need the `word-break` property as the below - as recommended in the article you've shared - don't seem to break an overflowing word (assuming `hyphens` doesn't work).

```
 overflow-wrap: break-word;
 word-wrap: break-word;
```
I'll update the code in a bit and can share a PR for testing/reviewing.
